packages/block-editor/src/hooks/utils.js

```diff
@@ -177,3 +184,76 @@ export function shouldSkipSerialization( blockType, featureSet, feature ) {
 
 	return skipSerialization;
 }
+
+export function useBlockSettings( name, parentLayout ) {
```

Comment:
Is it worth adding a doc block to this function? From reading through the changes, I wasn't sure what the intended scope is for settings included here — it looks like it's currently typography + spacing + layout, but not colors or position, for example. I imagine we'll continue to add to this function as each panel is extracted / consolidated, with the eventual goal for it to include all block settings?

Reply:
Yes, that's the goal I'm also thinking of making its result a locker prop in BlockEditContext, that way it's always available. Maybe I'll do the same for the "value" prop (the style extracted from all attributes) but I'm not 100% decided at this point.